git_machete/cmd.py

```diff
@@ -532,7 +532,7 @@ def add(self, b: str) -> None:
             print(fmt(f"Added branch `{b}` as a new root"))
         else:
             if not onto:
-                u = infer_upstream(self.cli_ctxt, b, condition=lambda x: x in self.managed_branches, reject_reason_message="this candidate is not a managed branch")
+                u = infer_upstream(self, self.cli_ctxt, b, condition=lambda x: x in self.managed_branches, reject_reason_message="this candidate is not a managed branch")
```

Comment:
`infer_upstream` to be pulled into `MacheteContext` (probably as private method?)